hudi-utilities/src/test/java/org/apache/hudi/utilities/sources/TestKafkaSource.java

```diff
@@ -242,7 +245,7 @@ public void testJsonKafkaSourceWithConfigurableUpperCap() {
     HoodieTestDataGenerator dataGenerator = new HoodieTestDataGenerator();
     TypedProperties props = createPropsForJsonSource(500L, "earliest");
 
-    Source jsonSource = new JsonKafkaSource(props, jsc, sparkSession, schemaProvider);
+    Source jsonSource = new JsonKafkaSource(props, jsc, sparkSession, schemaProvider, metrics);
     SourceFormatAdapter kafkaSource = new SourceFormatAdapter(jsonSource);
 
     // 1. Extract without any checkpoint => get all the data, respecting sourceLimit
```

Comment:
@liujinhui1994 thanks for your contribution, I left some comments you can consider. 
Besides, I have an idea here, it should be more elegant if you can init the metric inside kafka source(json, avro) without breaking the constructor of `org.apache.hudi.utilities.sources.Source` cc @yanghua WDYT?

Reply:
It seems we can use the same metric in one `sync` loop.  how about initializing it at the begging(in `DeltaSync` ) and deliver it to the source constructor?

Reply:
This is also a way. Currently non-kafka souce has not used metic, so I made a judgment and only passed metic when kafka souce